packages/command/src/sfdxFlags.ts

```diff
@@ -115,11 +125,43 @@ function buildVersion(options?: flags.BaseBoolean<boolean>): flags.Discriminated
 
 // sfdx
 
-function buildArray<T>(options: flags.Array<T>): flags.Discriminated<flags.Array<T>> {
-  return option('array', options, val => {
-    const vals = val.split(options.delimiter || ',');
-    return options.map ? vals.map(options.map) : vals;
-  });
```

Comment:
This previous impl was much simpler, but didn't handle `options` (akak `enum` value constraints) in arrays and had some typing problems in tsc 3.2.2 that are fixed now.